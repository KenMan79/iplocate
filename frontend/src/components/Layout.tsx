import React from "react";
import styled, { createGlobalStyle } from "styled-components";
import { Footer } from "./Footer";

export const Layout: React.FC = (props) => {
  const { children } = props;

  return (
    <>
      <Global />
      <Wrapper>
        <Content>{children}</Content>
        <Footer />
      </Wrapper>
    </>
  );
};

const Global = createGlobalStyle`
  * {
    box-sizing: border-box;
  }
  html, body {
    padding: 0;
    margin: 0;
  }
`;

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  min-height: 100vh;
  font-family: var(--font-family);
`;

const Content = styled.div`
  flex: 1 1 auto;
  background-color: var(--background-main);
`;
