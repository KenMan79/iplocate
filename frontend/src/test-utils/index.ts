export * from "./test-utils";
export * from "./ipdata";
