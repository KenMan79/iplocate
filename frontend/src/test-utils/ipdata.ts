import { IP } from "types";

export const baseIP: IP = {
  continent: {
    code: "EU",
    geonameId: 6255148,
    names: {
      en: "Europe",
    },
  },
  country: {
    geonameId: 3017382,
    isInEuropeanUnion: true,
    isoCode: "FR",
    names: {
      en: "France",
    },
  },
  traits: {
    isAnonymous: false,
    isAnonymousProxy: false,
    isAnonymousVpn: false,
    isHostingProvider: false,
    isLegitimateProxy: false,
    isPublicProxy: false,
    isResidentialProxy: false,
    isSatelliteProvider: false,
    isTorExitNode: false,
    ipAddress: "2.2.2.2",
    network: "2.2.0.0/18",
  },
  city: {},
  location: {
    accuracyRadius: 500,
    latitude: 48.8582,
    longitude: 2.3387,
    timeZone: "Europe/Paris",
  },
};
