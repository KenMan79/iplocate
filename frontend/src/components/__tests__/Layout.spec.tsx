import { get } from "https";
import React from "react";
import { render } from "test-utils";
import { Layout } from "../Layout";

jest.mock("../Footer", () => ({
  Footer: () => <footer>Mock Footer</footer>,
}));

afterAll(() => {
  jest.restoreAllMocks();
});

it("matches snapshot", () => {
  const { asFragment } = render(<Layout />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders children", () => {
  const { getByTestId } = render(
    <Layout>
      <div data-testid="child">Content</div>
    </Layout>
  );
  expect(getByTestId("child")).toHaveTextContent("Content");
});
