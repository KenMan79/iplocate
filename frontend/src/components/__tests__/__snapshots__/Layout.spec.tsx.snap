// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  height: 100vh;
  font-family: var(--font-family);
}

.c1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  position: relative;
  background-color: var(--background-main);
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <footer>
      Mock Footer
    </footer>
  </div>
</DocumentFragment>
`;
